test/DynamoCoreTests/SerializationTests.cs

```diff
@@ -84,31 +84,34 @@ public WorkspaceComparisonData(WorkspaceModel workspace, EngineController contro
                 {
                     NodeTypeMap.Add(n.GUID, n.GetType());
 
-                    var portvalues = new List<object>();
-                    foreach (var p in n.OutPorts)
-                    {
-                        var value = n.GetValue(p.Index, controller);
-                        if (value.IsCollection)
-                        {
-                            portvalues.Add(GetStringRepOfCollection(value));
-                        }
-                        else
-                        {
-                            portvalues.Add(value.StringData);
-                        }
-                    }
-                    
+                    var portvalues = n.OutPorts.Select(p =>
+                        GetDataOfValue(n.GetValue(p.Index, controller))).ToList<object>();
+
                     NodeDataMap.Add(n.GUID, portvalues);
                     InportCountMap.Add(n.GUID, n.InPorts.Count);
                     OutportCountMap.Add(n.GUID, n.OutPorts.Count);
                 }
             }
         }
 
-        private static string GetStringRepOfCollection(ProtoCore.Mirror.MirrorData collection)
+        private static object GetDataOfValue(ProtoCore.Mirror.MirrorData value)
         {
-            var items = string.Join(",", collection.GetElements().Select(x => x.IsCollection ? GetStringRepOfCollection(x) : x.StringData));
-            return "{" + items + "}";
+            if (value.IsCollection)
+            {
+                return value.GetElements().Select(x => GetDataOfValue(x)).ToList<object>();
+            }
+
+            if (!value.IsPointer)
+            {
+                var data = value.Data;
+
+                if (data != null)
+                {
+                    return data;
```

Comment:
I see the change to output values instead of strings. The serialization will most likely fail if that value is geometry, because geometry doesn't do string serialization very well :) Is there a plan to catch that later down the line?